pal/src/thread/pal_thread.cpp

```diff
@@ -191,6 +191,7 @@ static void InternalEndCurrentThreadWrapper(void *arg)
 --*/
 BOOL TLSInitialize()
 {
+    free_threads_spinlock.Reset();
```

Comment:
>free_threads_spinlock.Reset(); [](start = 4, length = 30)

Why you need to call "Reset()" for this one? Doesn't the constructor work? Or this could be called multiple times? If the spinlock is already Entered by some thread, wouldn't this call corrupts the existing data?